docker/incremental_load.sh.tpl

```diff
@@ -22,38 +22,96 @@ RUNFILES="${PYTHON_RUNFILES:-${BASH_SOURCE[0]}.runfiles}"
 
 DOCKER="${DOCKER:-docker}"
 
-# List all images identifier (only the identifier) from the local
-# docker registry.
-IMAGES="$("${DOCKER}" images -aq)"
+# Fetch the diff ids of the layers loaded in the docker daemon already
+IMAGES=$(${DOCKER} inspect $(${DOCKER} images -aq) |& \
+  grep sha256 | grep -v Id | cut -d'"' -f 2 | cut -d':' -f 2 | \
+  sort | uniq)
 IMAGE_LEN=$(for i in $IMAGES; do echo -n $i | wc -c; done | sort -g | head -1 | xargs)
 
 [ -n "$IMAGE_LEN" ] || IMAGE_LEN=64
 
-function incr_load() {
+# Create temporary files in which to record things to clean up.
+TEMP_FILES=$(mktemp)
+TEMP_IMAGES=$(mktemp)
+function cleanup() {
+  cat "${TEMP_FILES}" | xargs rm -rf> /dev/null 2>&1 || true
+  cat "${TEMP_IMAGES}" | xargs "${DOCKER}" rmi > /dev/null 2>&1 || true
+
+  rm -rf "${TEMP_FILES}"
+  rm -rf "${TEMP_IMAGES}"
+}
+trap cleanup EXIT
+
+
+function load_legacy() {
```

Comment:
This operation will be really slow if the base image is say ubuntu compressed, and we redo it at every run.